privatek8s.tf

```
@@ -169,6 +182,55 @@ resource "azurerm_kubernetes_cluster_node_pool" "infraciarm64" {
   tags = local.default_tags
 }
 
+# nodepool dedicated for the infra.ci.jenkins.io controller
+resource "azurerm_kubernetes_cluster_node_pool" "infracictrlarm64" {
+  name                  = "arm64infracictrl"
+  vm_size               = "Standard_D4pds_v5" # 4 vCPU, 16 GB RAM, local disk: 150 GB and 19000 IOPS
+  os_disk_type          = "Ephemeral"
+  os_disk_size_gb       = 150 # Ref. Cache storage size at https://learn.microsoft.com/en-us/azure/virtual-machines/dpsv5-dpdsv5-series#dpdsv5-series (depends on the instance size)
+  orchestrator_version  = local.kubernetes_versions["privatek8s"]
+  kubernetes_cluster_id = azurerm_kubernetes_cluster.privatek8s.id
+  enable_auto_scaling   = true
+  min_count             = 1
+  max_count             = 3
+  zones                 = [1] # zone 1 is dedicated to ARM64 by azure
+  vnet_subnet_id        = data.azurerm_subnet.privatek8s_infra_ci_controller_tier.id
+
+  node_taints = [
+    "jenkins=infra.ci.jenkins.io:NoSchedule",
+    "kind=infra.ci.jenkins.io"
+  ]
+  lifecycle {
+    ignore_changes = [node_count]
+  }
+
+  tags = local.default_tags
+}
+
+# nodepool dedicated for the release.ci.jenkins.io controller
+resource "azurerm_kubernetes_cluster_node_pool" "releasecictrlarm64" {
+  name                  = "arm64releasecictrl"
+  vm_size               = "Standard_D4pds_v5" # 4 vCPU, 16 GB RAM, local disk: 150 GB and 19000 IOPS
+  os_disk_type          = "Ephemeral"
+  os_disk_size_gb       = 150 # Ref. Cache storage size at https://learn.microsoft.com/en-us/azure/virtual-machines/dpsv5-dpdsv5-series#dpdsv5-series (depends on the instance size)
+  orchestrator_version  = local.kubernetes_versions["privatek8s"]
+  kubernetes_cluster_id = azurerm_kubernetes_cluster.privatek8s.id
+  enable_auto_scaling   = true
+  min_count             = 1
+  max_count             = 3
+  zones                 = [1] # zone 1 is dedicated to ARM64 by azure
+  vnet_subnet_id        = data.azurerm_subnet.privatek8s_release_ci_controller_tier.id
+
+  node_taints = [
+    "jenkins=release.ci.jenkins.io:NoSchedule",
+    "kind=release.ci.jenkins.io"
+  ]
```

```suggestion
  node_taints = [
    "jenkins=release.ci.jenkins.io:NoSchedule",
    "jenkins-component=controller:NoSchedule"
  ]
```

- As per https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/, the taint must have a 3rd elements which is the "Taint effect": adding `: NoSchedule` at the end
- Using `jenkins-component` instead of `kind`: the latter is too generic to have a clear meaning. I propose `jenkins-component` as we keep using "agent / non agent" wording: if we stick to "controller / agent" official wording that will help to make thing clear in the context of a "Jenkins" installation

WDYT?